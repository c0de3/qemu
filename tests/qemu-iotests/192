#!/usr/bin/env python
#
# Copyright (C) 2017 Red Hat, Inc.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
# Author: Kashyap Chamarthy <kchamart@redhat.com>
#         [Converted to Python from the original Bash version by Fam
#         Zheng (<famz@redhat.com)]
#
# Purpose: Test NBD export with '-incoming' (non-shared storage
# migration use case from libvirt)

import os
import atexit
import iotests

iotests.verify_platform(['linux'])

img_size = '1G'
test_img_path = os.path.join(iotests.test_dir, 'dest.img')
iotests.qemu_img_pipe('create', '-f', iotests.imgfmt, test_img_path, img_size)

iotests.log('Launching VM...')
nbd_sock_path = os.path.join(iotests.test_dir, 'nbd.sock')
dest_vm = (iotests.VM('dest').add_drive(test_img_path)
                               .add_incoming('defer'))
dest_vm.launch()
atexit.register(dest_vm.shutdown)

iotests.log('Launching NBD server on destination...')
iotests.log(dest_vm.qmp('nbd-server-start', addr={'type': 'unix', 'data': {'path': nbd_sock_path}}))
iotests.log('Exporting the block device to NBD server...')
iotests.log(dest_vm.qmp('nbd-server-add', device='drive0', writable=True))
iotests.log('Stopping the NBD server on destination...')
iotests.log(dest_vm.qmp('nbd-server-stop'))
