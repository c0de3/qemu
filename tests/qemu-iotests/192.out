Launching VM...
Launching NBD server on destination...
{u'return': {}}
Exporting the block device to NBD server...
{u'return': {}}
Stopping the NBD server on destination...
{u'return': {}}
